uxarray/grid/grid.py

```diff
@@ -1390,3 +1396,39 @@ def isel(self, **dim_kwargs):
             raise ValueError(
                 "Indexing must be along a grid dimension: ('n_node', 'n_edge', 'n_face')"
             )
+
+    def compute_dual(self, method="global"):
+        """Compute the dual mesh for a grid, returns a new grid object.
+
+         Parameters
+        ----------
+        method: str, default="global"
+            Method for constructing the dual mesh, either "global" or "local"
+
+        Returns:
+        --------
+        dual : Grid
+            Dual Mesh Grid constructed
+        """
+
+        if method == "local":
+            raise ValueError("Local Dual Mesh is not yet supported, use global")
```

Comment:
Can you elaborate on what a "local" dual mesh would be?

Reply:
I will have to look into it more, but I added it because in tempest remap for meshes with holes, it uses a local dual mesh construction instead. My dual mesh algorithm supports meshes with holes though so maybe this isn't needed? @paullric maybe could give clarification on what a local dual is and why it's needed?